eZ/Bundle/EzPublishIOBundle/Migration/FileMigrator/FileMigrator.php

```diff
@@ -16,6 +16,21 @@ final class FileMigrator extends MigrationHandler implements FileMigratorInterfa
 {
     public function migrateFile(BinaryFile $binaryFile)
     {
+        if (!$this->migrateBinaryFile($binaryFile) || !$this->migrateMetadata($binaryFile)) {
+            return false;
+        }
+
+        $this->logInfo("Successfully migrated: '{$binaryFile->id}'");
+
+        return true;
+    }
+
+    private function migrateBinaryFile(BinaryFile $binaryFile): bool
+    {
+        if (get_class($this->fromBinarydataHandler) === get_class($this->toBinarydataHandler)) {
```

Comment:
These are supposed to be service instances, so you can compare if they are the same instance. Don't need - and even should not - to compare classes.

This is even more important since those service can have different configurations, which would not be detected by `get_class()` comparison.

```suggestion
        if ($this->fromBinarydataHandler === $this->toBinarydataHandler) {
```